nsecx.go

```diff
@@ -63,8 +63,10 @@ func (rr *NSEC3) Cover(name string) bool {
 	}
 
 	nextHash := rr.NextDomain
-	if ownerHash == nextHash { // empty interval
-		return false
+
+	// if empty interval found, try cover wildcard hashes so nameHash shouldn't match with ownerHash
```

Comment:
Maybe:
If empty interval is found, ...
And then I'm lost; what is the second check proving here? `nameHash` is not equal to `ownerHash` because there may be a wildcard?

[I think it's important to get this right, because NSEC3 is so complex]

Reply:
No, be more specific on what this checks does in the context of RFC 5155 and be a bit verbose about it. What does this check do and why is it needed (in what nsec3 corner case)

Reply:
RFC 5155 is not crystal clear about this. I read whole document but I couldn't explain correctly. I checked other implementations, like PERL libs, PowerDNS etc and debugged NSEC3 records and found this solution. PowerDNS has same implementation and there is no comment.

https://github.com/PowerDNS/pdns/blob/master/pdns/validate.cc#L30